docs/modules/data-structures/pages/vector-collections.adoc

```diff
@@ -0,0 +1,472 @@
+                <dimension>2</dimension>
+                <metric>DOT</metric>
+                <max-degree>10</max-degree>
+                <ef-construction>20</ef-construction>
+                <use-deduplication>false</use-deduplication>
+            </index>
+        </indexes>
+    </vector-collection>
+</hazelcast>
+----
+--
+YAML::
++
+--
+[source,yaml]
+----
+hazelcast:
+  vector-collection:
+    vector-collection-name:
+      indexes:
+        - name: index-1
+          dimension: 2
+          metric: DOT
+        - name: index-2
+          dimension: 2
+          metric: DOT
+          max-degree: 10
+          ef-construction: 20
+          use-deduplication: false
+----
+--
+Java::
++
+--
+[source,java]
+----
+Config config = new Config();
+VectorCollectionConfig collectionConfig = new VectorCollectionConfig("vector-collection-name")
+    .addVectorIndexConfig(
+            new VectorIndexConfig()
+                .setName("index-1")
+                .setDimension(2)
+                .setMetric(Metric.DOT)
+    ).addVectorIndexConfig(
+            new VectorIndexConfig()
+                .setName("index-2")
+                .setDimension(2)
+                .setMetric(Metric.DOT)
+                .setMaxDegree(10)
+                .setEfConstruction(20)
+                .setUseDeduplication(false)
+    );
+config.addVectorCollectionConfig(collectionConfig);
+----
+--
+====
+
+== Manage collection
+
+=== Get vector collection
+To get the collection, use one of the static methods of `VectorCollection`. Both methods will either create a new vector collection or return an existing one corresponding to the requested name.
+
+* Method `getCollection(HazelcastInstance instance, VectorCollectionConfig collectionConfig)`.
+** If a collection with the provided name does not exist, a new collection will be created with the given configuration. If the configuration for the collection already exists, the provided configuration should match the one created earlier; otherwise, an error will be thrown.
+** If a collection with the same name and configuration already exists, it will be returned.
+** If a collection with the same name but a different configuration exists, an error will be thrown.
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection vectorCollection = VectorCollection.getCollection(hazelcastInstance, vectorCollectionConfig);
+----
+--
+====
+
+* Method `getCollection(HazelcastInstance instance, VectorCollectionConfig collectionConfig)`.
+** If a collection with the provided name does not exist, the system will create the collection with the configuration created implicitly during static or dynamic configuration of the cluster. If the configuration does not exist, an error will be thrown.
+** If a collection with the provided name exists, it will be returned.
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
```

Comment:
this example is too abstract with `collectionName`. I like more the style used e.g. in IMap docs with concrete names, data etc. In concrete example we can also parameterize the generics.

Reply:
7e1b5c3452fd6c95ac9cd6d02014415617138c44